app/views/contexts/questionnaire_summary_context.py

```diff
@@ -19,17 +19,25 @@ def __call__(self, collapsible=True, answers_are_editable=True):
 
     def _build_all_groups(self):
         """ NB: Does not support repeating sections """
-        section_summary_context = SectionSummaryContext(
-            self._language,
-            self._schema,
-            self._answer_store,
-            self._list_store,
-            self._progress_store,
-            self._metadata,
-        )
-
         for section_id in self._router.enabled_section_ids:
-            for group in section_summary_context(Location(section_id=section_id))[
-                "summary"
-            ]["groups"]:
-                yield group
+            section = self._schema.get_section(section_id)
+            if section.get("summary", {}).get("items"):
```

Comment:
Not sure on this one, I understand we wish to hide lists in the Final summary, but if a list shares a group with another block then that will also be removed. In test_list_collector there is a block called test-number-block which is just a general question, this won't appear on the summary when I believe it should. 

N.B. This might be a pattern we shouldn't be using (i.e lists sharing groups) rather than a bug

Reply:
Yeah I spoke to @ajmaddaford about this - it's expected that any questions in a section that has a custom summary definition won't display those questions on the final summary

Reply:
Just to be clear, this is because we only use this in who lives here and don't have any requirements for any other behaviour. When we know more, we can build it.